org.adempiere.base/src/org/compiere/model/MOrder.java

```diff
@@ -746,7 +746,10 @@ public void setBPartner (MBPartner bp)
 		ss = bp.getInvoiceRule();
 		if (ss != null)
 			setInvoiceRule(ss);
-		ss = bp.getPaymentRule();
+		if (isSOTrx())
+			ss = bp.getPaymentRule();
+		else
+			ss = bp.getPaymentRulePO();
```

Comment:
I think the usual rule is if getPaymentRulePO() is null, we fall back to getPaymentRule().

Reply:
done